lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureCloudBlobClient.cs

```diff
@@ -67,14 +68,30 @@ public ICloudBlockBlob GetBlockBlobReference(Uri uri)
             return new AzureCloudBlockBlob(uri, _client.Credentials);
         }
 
-        public BlobResultSegment ListBlobsSegmented(string prefix, bool useFlatListing, BlobListingDetails blobListingDetails,
-            int? maxResults, BlobContinuationToken continuationToken, BlobRequestOptions blobRequestOptions,
+        public BlobResultSegment ListBlobsSegmented(
+            string containerName,
+            string relativeAddress,
+            bool useFlatListing,
+            BlobListingDetails blobListingDetails,
+            int? maxResults,
+            BlobContinuationToken continuationToken,
+            BlobRequestOptions blobRequestOptions,
             OperationContext operationContext)
         {
-            var task = _client.ListBlobsSegmentedAsync(prefix, useFlatListing, blobListingDetails, maxResults,  continuationToken, 
-                                                        blobRequestOptions, operationContext);
-            task.Wait();
-            return task.Result;
+            CloudBlobContainer container = _client.GetContainerReference(containerName);
+            CloudBlobDirectory directory = container.GetDirectoryReference(relativeAddress);
+            return directory.ListBlobsSegmentedAsync(
+                useFlatListing,
+                blobListingDetails,
+                maxResults,
+                continuationToken,
+                blobRequestOptions,
+                operationContext).GetAwaiter().GetResult();
```

Comment:
`.GetAwaiter().GetResult()` can be simply `.Result`

Reply:
I see some discussions on this topic and it appears the two have different responses if an exception is thrown: [https://stackoverflow.com/questions/17284517/is-task-result-the-same-as-getawaiter-getresult](https://stackoverflow.com/questions/17284517/is-task-result-the-same-as-getawaiter-getresult)

From that discussion, I wager that GetAwaiter().GetResult() is the better option to use.

---
In reply to: [196655236](https://github.com/apache/reef/pull/1464#discussion_r196655236) [](ancestors = 196655236)

Reply:
good point!